UPGRADING-HADOOP.md

```diff
@@ -0,0 +1,181 @@
+
+
+    version=2.7.2
+    juju set namenode hadoop_version=$version
+    juju set slave hadoop_version=$version
+2. Prepare the rolling upgrade:
+
+
+    juju action do namenode/0 hadoop-upgrade version=$version prepare=true
+3. Upgrade both namenodes:
+
+
+    juju action do namenode/0 hadoop-upgrade version=$version
+    juju action do namenode/1 hadoop-upgrade version=$version 
+4. Upgrade datanodes:
+
+
+    juju action do slave/0 hadoop-upgrade version=$version
+    juju action do slave/1 hadoop-upgrade version=$version
+    juju action do slave/2 hadoop-upgrade version=$version
+
+If data persistence is important to you, you should upgrade a subset of your
+slaves/datanodes, perform some tests (e.g. check existing data) and then
+continue with the upgrade or rollback. 
+
+Once you have have finished upgrading your units (successfully or not) you will
+need to do one of the following:
+
+1. finalize - cleans up the HDFS filesystem, and marks the rolling upgrade as
+complete
+2. downgrade - restores a previous version of hadoop (after which you ***must
+still finalize***)
+3. rollback - **not yet available** will restore a previous version of hadoop
+and restore HDFS file system to the state it was in when 'prepare' was run. 
+
+
+### Upgrade (standalone)
+
+  While we do not officially support the upgrade of a standalone deployment, it is possible using the following steps:
+
+  1. Config set version
+  2. Prepare rolling upgrade
+  3. Upgrade datanodes and then namenode using the standalone=true parameter
+
+
+### Finalize
+
+This is a fairly straight forward action and can be run on only one of the
+namenodes - this restarts the namenode daemons in 'normal' mode, but we
+recommend running it on both your namenodes to force those daemons to restart: 
+
+  juju action do namenode/$a hadoop-upgrade version=$version postupgrade=finalize
+
+One reason finalize should be run is that any files deleted after 'prepare' and
+before 'finalize' will remain (hidden) in your HDFS filesystem - therefore deleting
+files will not free up space - this is to enable a rollback. 
+
+
+### Downgrade
+
+Slightly more actions are required to downgrade HDFS than to finalize, and the
+recommended process is:
+
+1. Set hadoop_version config value for each service:
+
+
+    version=2.7.1
+    juju set namenode hadoop_version=$version
+    juju set slave hadoop_version=$version
+2. Downgrade datanodes:
+
+
+    juju action do slave/0 hadoop-upgrade version=$version postupgrade=downgrade
+    juju action do slave/1 hadoop-upgrade version=$version postupgrade=downgrade
+    juju action do slave/2 hadoop-upgrade version=$version postupgrade=downgrade
+    
+3. Downgrade both namenodes:
+
+
+    juju action do namenode/0 hadoop-upgrade version=$version postupgrade=downgrade
+    juju action do namenode/1 hadoop-upgrade version=$version postupgrade=downgrade
+
+***Once the downgrade is complete, you must still finalize the rolling upgrade***
+
+
+### Rollback
```

Comment:
I am not a fan of mentioning a feature we do not have yet. What if we never get to work on it?